### ISSUE TITLE
Null-island location when start tip sesstion

### ISSUE THREAD

**Issue by RingerJK** [mapbox/mapbox-navigation-android#4581]
`LocationObserver` might spread `null-island` location(lon, lat are 0.0) which is mean NN hasn't received locations update. 

We may handle it on the SDK side: cache locations for users and set it instead `null-island`. Also, if no cached locations and NN is trying to set `null-island` we can skip location update as it's not useful

cc @LukasPaczos @kmadsen

**Comment by Zayankovsky** [comment]
Hey @mapbox/navnative, is it possible for NavigationObserver to provide invalid location like Null island, if no valid locations were received so far? I believe it was possible with the now deprecated `Navigator.getStatus(long)` function.

**Comment by Guardiola31337** [comment]
Are you returning `INVALID` `RouteState` (from `NavigationStatus`) in such scenarios? Wondering if you should include that information as part of `FixLocation` or even `NavigationStatusOrigin` 🤔 

Linking https://github.com/mapbox/mapbox-navigation-android/pull/4391 for reference.

**Comment by mskurydin** [comment]
@Zayankovsky could you elaborate a bit more on the situation you are observing?

**Comment by Zayankovsky** [comment]
@mskurydin previously we used `Navigator.getStatus(long monotonicTimestampNanoseconds)` function to retrieve enhanced location. If there was no previous `Navigator.updateLocation` call, then this function would return invalid location with both latitude and longitude set to zero. Recently we switched to `NavigatorObserver`, which we set via `Navigator.addObserver` call. So I was wondering, if it is still possible to receive invalid location that way, for example if you add the observer before any `Navigator.updateLocation` calls are made.
